src/plugins/intel_npu/src/plugin/npuw/weights_bank.cpp

```diff
@@ -222,7 +222,8 @@ void Bank::read_and_add_tensor(std::istream& stream, int64_t uid, const std::str
     auto iter_device = device_bank.storage.find(uid);
 
     if (iter_device != device_bank.storage.end()) {
-        // Already allocated
+        // Shouldn't be possible
+        NPUW_ASSERT(false);
         return;
     }
```

Comment:
It's a change in the contract here, what was OK previously is now not OK, can you please explain more here?

Reply:
It occurs during bank deserialization. Our storage is an unordered_map with unique UIDs which all have been evaluated and serialized at this point. Thus it shouldn't be possible to encounter a duplicate UIDs during deserialization. Just a sanity check - this if can be removed altogether with no effect